src/meta/src/controller/streaming_job.rs

```diff
@@ -406,6 +418,251 @@ impl CatalogController {
         Ok(())
     }
 
+    pub async fn create_job_catalog_for_replace(
+        &self,
+        streaming_job: &StreamingJob,
+        ctx: &StreamContext,
+        version: &PbTableVersion,
+    ) -> MetaResult<ObjectId> {
+        let id = streaming_job.id();
+        let inner = self.inner.write().await;
+        let txn = inner.db.begin().await?;
+
+        // 1. check version.
+        let original_version: Option<TableVersion> = Table::find_by_id(id as TableId)
+            .select_only()
+            .column(table::Column::Version)
+            .into_tuple()
+            .one(&txn)
+            .await?
+            .ok_or_else(|| MetaError::catalog_id_not_found(ObjectType::Table.as_str(), id))?;
+        let original_version = original_version.expect("version for table should exist");
+        if version.version != original_version.inner_ref().version + 1 {
+            return Err(MetaError::permission_denied("table version is stale"));
+        }
+
+        // 2. create streaming object for new replace table.
+        let obj_id = Self::create_streaming_job_obj(
```

Comment:
Here we still use a dummy id that represents a new streaming job to achieve table replacement. The dummy streaming job will alway be `Foreground`, so that if recovery encountered and the replacement is not completed it will be cleaned.